### PR TITLE
Avoid optimization data copying

### PR DESCRIPTION
**Pull request by ViktorHofer**
When restoring optimization data packages for libraries we don't need
to copy their contents into our artifacts directory.

@ericstj I didn't see a reason why we need to copy the optimization data content around. Please let me know if I'm missing something. This removes some unnecessary work happening in the builds of the other sub-repos.

### CODE REVIEW

**Comment by ericstj** [comment]
@ViktorHofer previously we needed to copy because the convention expected by the optimization tooling didn't match the structure in the package.  Double check that this now matches.

File: eng/codeOptimization.targets [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
           Condition="'$(IsEligibleForNgenOptimization)' == 'true'"
           BeforeTargets="CoreCompile">
     <PropertyGroup>
-      <IbcOptimizationDataDir Condition="'$(OSGroup)' == 'Unix' or '$(OSGroup)' == 'Linux'">$(IbcOptimizationDataDir)$(LinuxCoreFxOptimizationDataPackage)\</IbcOptimizationDataDir>
-      <IbcOptimizationDataDir Condition="'$(OSGroup)' != 'Unix' and '$(OSGroup)' != 'Linux'">$(IbcOptimizationDataDir)$(WindowsCoreFxOptimizationDataPackage)\</IbcOptimizationDataDir>
+      <IbcOptimizationDataDir Condition="'$(OSGroup)' == 'Unix' or '$(OSGroup)' == 'Linux'">$(Pkgoptimization_linux-x64_IBC_CoreFx)\</IbcOptimizationDataDir>
+      <IbcOptimizationDataDir Condition="'$(OSGroup)' != 'Unix' and '$(OSGroup)' != 'Linux'">$(Pkgoptimization_windows_nt-x64_IBC_CoreFx)\</IbcOptimizationDataDir>
     </PropertyGroup>
     <ItemGroup>
       <_optimizationDataAssembly Include="$(IbcOptimizationDataDir)**\$(TargetFileName)" />
```
Comment (ericstj) [inline eng/codeOptimization.targets]: You need to add a condition here, `'$(IbcOptimizationDataDir)' != ''`

**Comment by stephentoub** [comment]
@ViktorHofer, are you still working on this?

**Comment by ViktorHofer** [comment]
I'll revisit this when I have time.

**Pull request closed without merging**